### PR TITLE
First Tipers

### PR DESCRIPTION
**Pull request by oliviaoddo**
- add contribution issues for first tipers
- webhook fires when a branch is created that is prefixed with `first-tipers-`
- **NOTE** be sure to enable the `create` webhook in the bot settings once this is merged

### CODE REVIEW

File: instructions.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+- [ ] 📝 **Update** the file [$FILENAME]($BRANCH_URL) in the `$REPO` repository (press the little pen Icon) and edit the line as shown below.
+
+
+``````diff
+$DIFF
+``````
+
+
+- [ ] 💾 **Commit** your changes
```
Comment (gdi2290) [inline instructions.md]: can you talk about how they should make their commitmessages

File: src/get-commit.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+function getCommit(state) {
+  state.debug(`getting commit "${state.sha}"`)
+
+  return state.api.repos
+    .getCommit({
+      owner: state.owner,
+      repo: state.installRepo,
+      sha: state.sha
+    })
+
```
Comment (gdi2290) [inline src/get-commit.js]: remove this gap

File: src/handle-create.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    template,
+    labels: config.labels,
+    sha: null,
+    repoDefaultBranch: repository.default_branch,
+    customTemplateUrl: config.template,
+    issueRepo: config.repository
+  }
+
+  return getBranch(state)
+    .then(getTemplateContent.bind(null, state))
```
Comment (gdi2290) [inline src/handle-create.js]: use arrow functions for these rather than bind
() => getTemplateContent(state)

File: src/handle-pr-change.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
+const fs = require('fs')
```
Comment (gdi2290) [inline src/handle-pr-change.js]: do we need this?

File: src/create-issue.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+function createIssue(state) {
+  const { filename, patch, branchUrl, message, authorLogin } = state.commit
+  const content = state.template
```
Comment (gdi2290) [inline src/create-issue.js]: create a function that interpolates the values into the template string

File: src/create-issue.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    .replace(/\$REPO/, state.installRepo)
+    .replace(/\$AUTHOR/, `@${authorLogin}`)
+
+  state.debug('creating issue...')
+
+  return state.api.issues
+    .create({
+      owner: state.owner,
+      repo: state.issueRepo,
+      title: message.split('\n\n')[0],
```
Comment (gdi2290) [inline src/create-issue.js]: move this above as `const title = message.split('\n\n')[0]`

File: src/delete-branch.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+function deleteBranch(state) {
+  return state.api.gitdata
+    .deleteReference({
+      owner: state.owner,
+      repo: state.installRepo,
+      ref: 'heads/' + state.branch
```
Comment (gdi2290) [inline src/delete-branch.js]: move this above as `const ref = 'heads/' + state.branch`

File: src/delete-branch.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    })
+    .then(result => {
+      state.debug(`branch deleted: heads/${state.branch}`)
+      return state
+    })
+    .catch(err => {
+      if (err.code === 403)
```
Comment (gdi2290) [inline src/delete-branch.js]: use brackets for if/else

File: src/get-commit.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  state.debug(`getting commit "${state.sha}"`)
+
+  return state.api.repos
+    .getCommit({
+      owner: state.owner,
+      repo: state.installRepo,
+      sha: state.sha
+    })
+    .then(function(result) {
+      const { filename, patch, blob_url: blobUrl } = result.data.files[0]
```
Comment (gdi2290) [inline src/get-commit.js]: will this error out if there aren't any files?

File: src/get-commit.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+      var branchUrl = blobUrl.replace(
```
Comment (gdi2290) [inline src/get-commit.js]: const

File: src/handle-create-event.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    issueRepo: config.repository
+  }
+
+  return getBranch(state)
+    .then(getTemplateContent)
+    .then(getCommit)
+    .then(createIssue)
+    .then(deleteBranch)
+    .catch(err => {
+      // log
```
Comment (gdi2290) [inline src/handle-create-event.js]: you were logging before. I just used // log as an example code

File: src/get-commit.js (hunk 9 above)
Comment (oliviaoddo) [reply src/get-commit.js]: yes

Comment (gdi2290) [reply src/get-commit.js]: anywhere that can error in these functions you need to check and return Promise.reject(new Error('error fields etc replace this text'))

**Pull request merged**